hydroengine_service/dgds_functions.py

```diff
@@ -0,0 +1,427 @@
+
+    return image
+
+def get_image_collection_info(source, start_date=None, end_date=None, image_num_limit=None):
+    """
+    Return list of objects with imageId and time for all images in an ImageCollection, or Image, source
+    :param source: String,
+    :param start_date: String, start date to filter collection on
+    :param end_date: String, end date to filter collection on
+    :param image_num_limit: limit of objects to return
+    :return: List of dictionaries
+    """
+    data_params = get_dgds_source_vis_params(source)
+    type = data_params.get('type', 'ImageCollection')
+
+    # Get images based on source requested
+    if type == 'ImageCollection':
+        collection = ee.ImageCollection(source)
+    elif type == 'Image':
+        collection = ee.ImageCollection.fromImages([ee.Image(source)])
+    else:
+        msg = f'Object of type {type} not supported.'
+        logger.debug(msg)
+        return
+
+    if end_date and not start_date:
+        msg = f'If endDate provided, must also include startDate'
+        logger.debug(msg)
+        return
+
+    if start_date:
+        start = ee.Date(start_date)
+        end = start.advance(1, 'day')
+        if end_date:
+            end = ee.Date(end_date)
+        collection = collection.filterDate(start, end)
+
+    n_images = collection.size().getInfo()
+    if not n_images:
+        msg = f'No images available between startDate={start_date} and endDate={end_date}'
+        logger.debug(msg)
+        return
+
+    if image_num_limit:
+        # get a limited number of latest images
+        collection = collection.limit(image_num_limit, 'system:time_start', False)
+
+    # Sort ascending
+    collection = collection.sort('system:time_start', True)
+
+    ids = ee.List(collection.aggregate_array('system:id')).getInfo()
+
+    dates = ee.List(collection.aggregate_array('system:time_start'))
+    if dates.length().getInfo() == 0:
+        date_list = [None] * len(ids)
+    else:
+        date_list = dates.map(lambda i: ee.Date(i).format()).getInfo()
+
+    response = []
+    for id, date in zip(ids, date_list):
+        object = {
+            'imageId': id,
+            'date': date
+        }
+        response.append(object)
+
+    return response
+
+def _get_wms_url(image_id, type='ImageCollection', band=None, function=None, min=None, max=None, palette=None):
+    """
+    Get WMS url from image_id
+    :param image_id: String, Google Earth Engine image id
+    :param type: String, type of source, either Image or ImageCollection
+    :param band: String, name of band in the image
+    :param function: String, function applied to the image
+    :param min: Float, minimum value of visualization
+    :param max: Float, maximum value of visualization
+    :param palette: List, palette applied to image visualization, given as list of hex codes.
+    :return: Dictionary, json object with image and wms info
+    """
+    # Specialized styling for GEBCO
+    if 'gebco'in image_id:
```

Comment:
There goes your generic function. ;)
Would at least make a TODO to improve this.